File: lib/app/data/models/drawer_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'item_unit.dart';

part 'drawer_item.freezed.dart';
part 'drawer_item.g.dart';

enum DrawerStatus {
  organized,
  unorganized;

  String get displayName {
    switch (this) {
      case DrawerStatus.organized:
        return 'Organized';
      case DrawerStatus.unorganized:
        return 'Unorganized';
    }
  }
}

@freezed
class DrawerItem with _$DrawerItem {
  const factory DrawerItem({
    required String id,
    @Default('') String category,
    required String name,
    @Default('') String location,
    @Default(DrawerStatus.unorganized) DrawerStatus status,
    DateTime? lastOrganized,
    String? notes,
    // Quantity and unit tracking
    @Default(0) int currentQuantity, // Current count (e.g., 3)
    @Default(0) int targetQuantity, // Target count (e.g., 5)
    @Default(ItemUnit()) ItemUnit unit, // Unit type (e.g., "pieces", "pairs", "t-shirts")
  }) = _DrawerItem;

  factory DrawerItem.fromJson(Map<String, dynamic> json) =>
      _$DrawerItemFromJson(json);
}

File: backend/app/models/drawer_item.py
"""DrawerItem model matching Flutter's drawer_item.dart."""

from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field

from .common import ItemUnit


class DrawerStatus(str, Enum):
    """Drawer item status."""
    organized = "organized"
    unorganized = "unorganized"


class DrawerItemBase(BaseModel):
    """Base drawer item fields."""
    name: str = Field(..., min_length=1, max_length=200)
    category: str = ""
    location: str = ""
    status: DrawerStatus = DrawerStatus.unorganized
    notes: str | None = None
    current_quantity: int = 0
    target_quantity: int = 0
    unit: ItemUnit = Field(default_factory=ItemUnit)
    styles: list[str] = Field(default_factory=list)


class DrawerItemCreate(DrawerItemBase):
    """Drawer item creation request."""
    pass


class DrawerItemUpdate(BaseModel):
    """Drawer item update request."""
    name: str | None = Field(None, min_length=1, max_length=200)
    category: str | None = None
    location: str | None = None
    status: DrawerStatus | None = None
    notes: str | None = None
    current_quantity: int | None = None
    target_quantity: int | None = None
    unit: ItemUnit | None = None
    styles: list[str] | None = None


class DrawerItem(DrawerItemBase):
    """Drawer item response model."""
    id: str
    user_id: str
    last_organized: datetime | None = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        use_enum_values = True


class DrawerStatusResponse(BaseModel):
    """Drawer status summary response."""
    total_items: int
    organized: int
    unorganized: int
    percentage: float
    last_organized: datetime | None

